packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -476,32 +475,8 @@ function build(syscall, forVatID, cacheSize, vatPowers, vatParameters) {
     return res => {
       harden(res);
       lsdebug(`ls.thenResolve fired`, res);
-
-      // We need to know if this is resolving to an imported/exported
-      // presence, because then the kernel can deliver queued messages. We
-      // could build a simpler way of doing this.
       const ser = m.serialize(res);
-      lsdebug(` ser ${ser.body} ${JSON.stringify(ser.slots)}`);
-      // find out what resolution category we're using
-      const unser = JSON.parse(ser.body);
-      if (
-        Object(unser) === unser &&
-        QCLASS in unser &&
-        unser[QCLASS] === 'slot'
-      ) {
-        const slot = ser.slots[unser.index];
-        insistVatType('object', slot);
-        syscall.fulfillToPresence(promiseID, slot);
-      } else {
-        // if it resolves to data, .thens fire but kernel-queued messages are
-        // rejected, because you can't send messages to data
-        syscall.fulfillToData(promiseID, ser);
-      }
-
-      // If we were *also* waiting on this promise (perhaps we received it as
-      // an argument, and also as a result=), then we are responsible for
-      // notifying ourselves. The kernel assumes we're a grownup and don't
-      // need to be reminded of something we did ourselves.
+      syscall.resolve([[promiseID, false, ser]]);
```

Comment:
ah, yeah, here the syscall signature is a three-element list, not a `[ vpid, { rejected, data } ]` 2-tuple, it'd be nice to be consistent (but as long as caller matches its callee, we can do that cleanup in a later PR)